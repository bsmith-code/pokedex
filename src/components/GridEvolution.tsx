import { IEvolutionChain } from 'pokeapi-typescript'

import { Box, Grid, ListItemButton, styled, Typography } from '@mui/material'

import { usePokemonContext } from 'context/PokemonContext'

import { flattenChain, getIdFromUrl, getPokemonSprite } from 'utils'

interface IProps {
  isFetching: boolean
  evolutions?: IEvolutionChain
}

const StyledGridItem = styled(ListItemButton)(({ theme }) => ({
  flexDirection: 'column',
  borderRadius: '4px',
  color: '#fff',
  padding: theme.spacing(4),
  backgroundColor: theme.palette.secondary.main,
  '&:hover': {
    backgroundColor: '#6704c9'
  }
}))

export const GridEvolution = ({ evolutions, isFetching }: IProps) => {
  const { handleChangePokemon } = usePokemonContext()
  const flattenedChain = flattenChain(evolutions?.chain)

  return !isFetching ? (
    <Box>
      <Typography variant="h6" sx={{ mb: 2 }}>
        Evolutions
      </Typography>
      {flattenedChain?.length ? (
        <Grid container spacing={2}>
          {flattenedChain.map(({ url, name }) => {
            const id = getIdFromUrl(url)
            return (
              <Grid item xs={12} sm={6} md={6} lg={4} key={`evolution-${name}`}>
                <StyledGridItem onClick={() => handleChangePokemon(name)}>
                  <img
                    alt={name}
                    src={getPokemonSprite(id)}
                    width="120px"
                    height="120px"
                    style={{ objectFit: 'contain' }}
                  />
                  <Typography variant="subtitle2">{name}</Typography>
                </StyledGridItem>
              </Grid>
            )
          })}
        </Grid>
      ) : (
        <Typography>No available evolutions</Typography>
      )}
    </Box>
  ) : (
    <Typography>Loading...</Typography>
  )
}
