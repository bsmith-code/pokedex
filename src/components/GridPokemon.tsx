import { useMemo, useState } from 'react'
import { INamedApiResource } from 'pokeapi-typescript'

import {
  Box,
  Grid,
  ListItemButton,
  Pagination,
  styled,
  Typography
} from '@mui/material'

import { useChangePokemon } from 'hooks/useSelectPokemon'

interface IProps {
  pokemon: INamedApiResource<void>[]
}

const count = 20

const StyledGridItem = styled(ListItemButton)`
  flex-direction: column;
  border-radius: 4px;
  border: 1px solid #efefef;
`

export const GridPokemon = ({ pokemon }: IProps) => {
  const { handleChangePokemon } = useChangePokemon()
  const [currentPage, setCurrentPage] = useState(1)

  const handleChangePage = (page: number) => {
    setCurrentPage(page)
  }

  const gridPage = currentPage - 1 // Grid is zero based
  const startIndex = gridPage * count
  const endIndex = startIndex + count

  const preparedPokemon = useMemo(
    () => pokemon.slice(startIndex, endIndex),
    [pokemon, startIndex, endIndex]
  )

  const getPokemonSprite = (id: number) =>
    `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${
      (id + 1) * currentPage
    }.png`

  return (
    <Box mt={4}>
      <Grid container spacing={2}>
        {preparedPokemon.map(({ name }, idx) => (
          <Grid item xs={12} sm={6} md={4} lg={3} key={`pokemon-grid-${name}`}>
            <StyledGridItem onClick={() => handleChangePokemon(name)}>
              <img alt={name} src={getPokemonSprite(idx)} />
              <Typography>{name}</Typography>
            </StyledGridItem>
          </Grid>
        ))}
      </Grid>
      <Box mt={6} display="flex" justifyContent="center">
        <Pagination
          page={currentPage}
          count={Math.floor(pokemon.length / count)}
          onChange={(_, page) => handleChangePage(page)}
        />
      </Box>
    </Box>
  )
}
