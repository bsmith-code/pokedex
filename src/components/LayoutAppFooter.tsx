import { Toolbar, Typography } from '@mui/material'

const LayoutAppFooter = () => {
  const year = new Date().getFullYear()

  return (
    <Toolbar
      variant="dense"
      component="footer"
      sx={{ bgcolor: 'primary.main', justifyContent: 'center' }}
    >
      <Typography color="#fff">
        &copy; Brian Matthew Smith, {year}. All Rights Reserved.
      </Typography>
    </Toolbar>
  )
}

export default LayoutAppFooter
