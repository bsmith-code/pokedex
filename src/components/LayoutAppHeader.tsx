import { useNavigate } from 'react-router-dom'

import { IconButton, Toolbar } from '@mui/material'

const LayoutAppHeader = () => {
  const navigate = useNavigate()

  const handleLogoClick = () => {
    navigate('/')
  }

  return (
    <Toolbar component="header" sx={{ bgcolor: 'primary.main' }}>
      <IconButton onClick={handleLogoClick}>
        <img
          alt="PrizePicks"
          src="/assets/logo.svg"
          width="140px"
          height="auto"
        />
      </IconButton>
    </Toolbar>
  )
}

export default LayoutAppHeader
