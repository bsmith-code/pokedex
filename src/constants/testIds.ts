export const TESTID_SEARCH_EMPTY = 'search-empty'
export const TESTID_SEARCH_HISTORY = 'search-history'

export const TESTID_DETAILS_ERROR = 'details-error'
export const TESTID_DETAILS_LOADING = 'details-loading'
export const TESTID_DETAILS_POKEMON = 'details-pokemon'
export const TESTID_DETAILS_RETURN = 'details-return'

export const TESTID_GRID_EVOLUTIONS = 'grid-evolutions'
export const TESTID_GRID_POKEMON = 'grid-pokemon'
export const TESTID_GRID_ERROR = 'grid-error'
export const TESTID_GRID_LOADING = 'grid-loading'
