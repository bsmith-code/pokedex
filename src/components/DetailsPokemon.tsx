import { useNavigate } from 'react-router-dom'

import ArrowBackIcon from '@mui/icons-material/ArrowBack'
import {
  Box,
  Card,
  CardContent,
  CardHeader,
  IconButton,
  Typography
} from '@mui/material'

import { usePokemonDetails } from 'hooks/usePokemonDetails'

import { GridEvolution } from 'components/GridEvolution'

import {
  TESTID_DETAILS_ERROR,
  TESTID_DETAILS_LOADING,
  TESTID_DETAILS_POKEMON,
  TESTID_DETAILS_RETURN
} from 'constants/testIds'

export const DetailsPokemon = () => {
  const {
    name,
    details,
    sprites,
    evolutions,
    speciesName,
    isErrorDetails,
    isFetchingDetails,
    isFetchingEvolutions
  } = usePokemonDetails()
  const navigate = useNavigate()

  const handleGoBack = () => {
    navigate('/')
  }

  if (isFetchingDetails) {
    return <div data-testid={TESTID_DETAILS_LOADING}>Loading...</div>
  }

  if (isErrorDetails) {
    return <div data-testid={TESTID_DETAILS_ERROR}>Invalid Pokemon.</div>
  }

  return (
    <>
      <Box display="flex" mt={2} data-testid={TESTID_DETAILS_POKEMON}>
        <Box flexShrink={0} position="relative">
          <IconButton
            onClick={handleGoBack}
            data-testid={TESTID_DETAILS_RETURN}
            sx={{ position: 'absolute', top: 0, left: 0 }}
          >
            <ArrowBackIcon />
          </IconButton>
          <img
            alt={name}
            src={sprites?.other?.['official-artwork']?.front_default ?? ''}
          />
        </Box>
        <Box p={2} flexGrow={1}>
          <Box mb={2}>
            <Typography component="h1" variant="h4">
              {name}
            </Typography>
            <Typography fontSize={14} color="grey.700" sx={{ mt: 1 }}>
              species: {speciesName}
            </Typography>
          </Box>
          {details.map(({ title, content }) => (
            <Card
              key={`details-${title}`}
              sx={{ bgcolor: 'secondary.main', color: 'common.white', mb: 2 }}
            >
              <CardHeader title={title} />
              <CardContent>{content}</CardContent>
            </Card>
          ))}
        </Box>
      </Box>
      <GridEvolution
        evolutions={evolutions}
        isFetching={isFetchingEvolutions}
      />
    </>
  )
}
