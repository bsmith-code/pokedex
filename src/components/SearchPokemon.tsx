import {
  Autocomplete,
  createFilterOptions,
  ListItemButton,
  TextField
} from '@mui/material'

import { usePokemonContext } from 'context/PokemonContext'

import { TESTID_SEARCH_POKEMON } from 'constants/testIds'

export const SearchPokemon = () => {
  const { pokemon, handleChangePokemon } = usePokemonContext()

  return (
    <Autocomplete
      data-testid={TESTID_SEARCH_POKEMON}
      filterOptions={createFilterOptions({ limit: 20 })}
      onChange={(_, value) => handleChangePokemon(value?.name ?? '')}
      options={pokemon}
      renderInput={params => <TextField {...params} label="Search Pokemon" />}
      getOptionLabel={({ name }) => name}
      renderOption={(props, { name }) => (
        <ListItemButton
          {...props}
          component="li"
          key={`pokemon-list-${name}`}
          data-testid={`pokemon-${name}`}
        >
          {name}
        </ListItemButton>
      )}
    />
  )
}
