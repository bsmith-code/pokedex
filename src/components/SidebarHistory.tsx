import { Box, List, ListItemButton, Paper, Typography } from '@mui/material'

import { usePokemonContext } from 'context/PokemonContext'

import { TESTID_SEARCH_EMPTY, TESTID_SEARCH_HISTORY } from 'constants/testIds'

export const SidebarHistory = () => {
  const { searchHistory, handleChangePokemon } = usePokemonContext()

  return (
    <Box
      px={4}
      py={3}
      flexShrink={1}
      flexBasis={300}
      component={Paper}
      bgcolor="grey.200"
      sx={{ '@media (max-width: 767px)': { order: 2, flexGrow: 1 } }}
    >
      <Typography variant="subtitle2">Search History</Typography>

      {searchHistory.length ? (
        <List data-testid={TESTID_SEARCH_HISTORY}>
          {searchHistory.map(name => (
            <ListItemButton
              component="li"
              key={`search-${name}`}
              onClick={() => handleChangePokemon(name)}
            >
              {name}
            </ListItemButton>
          ))}
        </List>
      ) : (
        <Typography data-testid={TESTID_SEARCH_EMPTY} sx={{ mt: 2 }}>
          No previous searches.
        </Typography>
      )}
    </Box>
  )
}
