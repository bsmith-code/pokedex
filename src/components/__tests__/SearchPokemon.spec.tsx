import { BrowserRouter, Route, Routes } from 'react-router-dom'
import { fireEvent, screen, waitFor, within } from '@testing-library/react'

import { PokemonContextProvider } from 'context/PokemonContext'

import { SearchPokemon } from 'components/SearchPokemon'

import { TESTID_SEARCH_POKEMON } from 'constants/testIds'

const MockComponent = () => (
  <BrowserRouter>
    <Routes>
      <Route
        index
        path="/pokemon?/:pokemonName?"
        element={
          <PokemonContextProvider>
            <SearchPokemon />
          </PokemonContextProvider>
        }
      />
    </Routes>
  </BrowserRouter>
)
describe('SearchPokemon', () => {
  it('correctly renders search options and navigates to chosen pokemon', async () => {
    const { baseElement } = customRender(<MockComponent />)

    const autocompleteEl = screen.getByTestId(TESTID_SEARCH_POKEMON)
    const buttonEl = within(autocompleteEl).getByTitle('Open')
    fireEvent.click(buttonEl)

    await waitFor(() => {
      expect(screen.getByRole('presentation')).toBeInTheDocument()
      expect(baseElement).toMatchSnapshot()
    })

    const optionEl = screen.getAllByRole('option')[4]
    fireEvent.click(optionEl)

    await waitFor(() => {
      expect(window.location.pathname).toBe(
        `/pokemon/${optionEl.textContent ?? ''}`
      )
    })
  })
})
