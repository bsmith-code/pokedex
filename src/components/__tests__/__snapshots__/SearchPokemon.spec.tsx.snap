// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPokemon correctly renders search options and navigates to chosen pokemon 1`] = `
<body>
  <div>
    <div
      aria-owns=":r0:-listbox"
      class="MuiAutocomplete-root Mui-expanded Mui-focused MuiAutocomplete-hasPopupIcon emotion-0"
      data-testid="search-pokemon"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root emotion-1"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-focused MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined emotion-2"
          data-shrink="true"
          for=":r0:"
          id=":r0:-label"
        >
          Search Pokemon
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot emotion-3"
        >
          <input
            aria-autocomplete="list"
            aria-controls=":r0:-listbox"
            aria-expanded="true"
            aria-invalid="false"
            autocapitalize="none"
            autocomplete="off"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused emotion-4"
            id=":r0:"
            role="combobox"
            spellcheck="false"
            type="text"
            value=""
          />
          <div
            class="MuiAutocomplete-endAdornment emotion-5"
          >
            <button
              aria-label="Close"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator MuiAutocomplete-popupIndicatorOpen emotion-6"
              tabindex="-1"
              title="Close"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-7"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root emotion-8"
              />
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline emotion-9"
          >
            <legend
              class="emotion-10"
            >
              <span>
                Search Pokemon
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiPopper-root MuiAutocomplete-popper emotion-11"
    data-popper-escaped=""
    data-popper-placement="bottom"
    data-popper-reference-hidden=""
    role="presentation"
    style="position: absolute; top: 0px; left: 0px; width: 0px; margin: 0px; transform: translate(0px, 0px);"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAutocomplete-paper emotion-12"
    >
      <ul
        aria-labelledby=":r0:-label"
        class="MuiAutocomplete-listbox emotion-13"
        id=":r0:-listbox"
        role="listbox"
      >
        <li
          aria-disabled="false"
          aria-selected="false"
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters MuiAutocomplete-option emotion-14"
          data-option-index="0"
          data-testid="pokemon-pokemon-0"
          id=":r0:-option-0"
          role="option"
          tabindex="-1"
        >
          pokemon-0
        </li>
        <li
          aria-disabled="false"
          aria-selected="false"
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters MuiAutocomplete-option emotion-14"
          data-option-index="1"
          data-testid="pokemon-pokemon-1"
          id=":r0:-option-1"
          role="option"
          tabindex="-1"
        >
          pokemon-1
        </li>
        <li
          aria-disabled="false"
          aria-selected="false"
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters MuiAutocomplete-option emotion-14"
          data-option-index="2"
          data-testid="pokemon-pokemon-2"
          id=":r0:-option-2"
          role="option"
          tabindex="-1"
        >
          pokemon-2
        </li>
        <li
          aria-disabled="false"
          aria-selected="false"
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters MuiAutocomplete-option emotion-14"
          data-option-index="3"
          data-testid="pokemon-pokemon-3"
          id=":r0:-option-3"
          role="option"
          tabindex="-1"
        >
          pokemon-3
        </li>
        <li
          aria-disabled="false"
          aria-selected="false"
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters MuiAutocomplete-option emotion-14"
          data-option-index="4"
          data-testid="pokemon-pokemon-4"
          id=":r0:-option-4"
          role="option"
          tabindex="-1"
        >
          pokemon-4
        </li>
        <li
          aria-disabled="false"
          aria-selected="false"
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters MuiAutocomplete-option emotion-14"
          data-option-index="5"
          data-testid="pokemon-pokemon-5"
          id=":r0:-option-5"
          role="option"
          tabindex="-1"
        >
          pokemon-5
        </li>
        <li
          aria-disabled="false"
          aria-selected="false"
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters MuiAutocomplete-option emotion-14"
          data-option-index="6"
          data-testid="pokemon-pokemon-6"
          id=":r0:-option-6"
          role="option"
          tabindex="-1"
        >
          pokemon-6
        </li>
        <li
          aria-disabled="false"
          aria-selected="false"
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters MuiAutocomplete-option emotion-14"
          data-option-index="7"
          data-testid="pokemon-pokemon-7"
          id=":r0:-option-7"
          role="option"
          tabindex="-1"
        >
          pokemon-7
        </li>
        <li
          aria-disabled="false"
          aria-selected="false"
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters MuiAutocomplete-option emotion-14"
          data-option-index="8"
          data-testid="pokemon-pokemon-8"
          id=":r0:-option-8"
          role="option"
          tabindex="-1"
        >
          pokemon-8
        </li>
        <li
          aria-disabled="false"
          aria-selected="false"
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters MuiAutocomplete-option emotion-14"
          data-option-index="9"
          data-testid="pokemon-pokemon-9"
          id=":r0:-option-9"
          role="option"
          tabindex="-1"
        >
          pokemon-9
        </li>
        <li
          aria-disabled="false"
          aria-selected="false"
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters MuiAutocomplete-option emotion-14"
          data-option-index="10"
          data-testid="pokemon-pokemon-10"
          id=":r0:-option-10"
          role="option"
          tabindex="-1"
        >
          pokemon-10
        </li>
        <li
          aria-disabled="false"
          aria-selected="false"
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters MuiAutocomplete-option emotion-14"
          data-option-index="11"
          data-testid="pokemon-pokemon-11"
          id=":r0:-option-11"
          role="option"
          tabindex="-1"
        >
          pokemon-11
        </li>
        <li
          aria-disabled="false"
          aria-selected="false"
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters MuiAutocomplete-option emotion-14"
          data-option-index="12"
          data-testid="pokemon-pokemon-12"
          id=":r0:-option-12"
          role="option"
          tabindex="-1"
        >
          pokemon-12
        </li>
        <li
          aria-disabled="false"
          aria-selected="false"
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters MuiAutocomplete-option emotion-14"
          data-option-index="13"
          data-testid="pokemon-pokemon-13"
          id=":r0:-option-13"
          role="option"
          tabindex="-1"
        >
          pokemon-13
        </li>
        <li
          aria-disabled="false"
          aria-selected="false"
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters MuiAutocomplete-option emotion-14"
          data-option-index="14"
          data-testid="pokemon-pokemon-14"
          id=":r0:-option-14"
          role="option"
          tabindex="-1"
        >
          pokemon-14
        </li>
        <li
          aria-disabled="false"
          aria-selected="false"
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters MuiAutocomplete-option emotion-14"
          data-option-index="15"
          data-testid="pokemon-pokemon-15"
          id=":r0:-option-15"
          role="option"
          tabindex="-1"
        >
          pokemon-15
        </li>
        <li
          aria-disabled="false"
          aria-selected="false"
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters MuiAutocomplete-option emotion-14"
          data-option-index="16"
          data-testid="pokemon-pokemon-16"
          id=":r0:-option-16"
          role="option"
          tabindex="-1"
        >
          pokemon-16
        </li>
        <li
          aria-disabled="false"
          aria-selected="false"
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters MuiAutocomplete-option emotion-14"
          data-option-index="17"
          data-testid="pokemon-pokemon-17"
          id=":r0:-option-17"
          role="option"
          tabindex="-1"
        >
          pokemon-17
        </li>
        <li
          aria-disabled="false"
          aria-selected="false"
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters MuiAutocomplete-option emotion-14"
          data-option-index="18"
          data-testid="pokemon-pokemon-18"
          id=":r0:-option-18"
          role="option"
          tabindex="-1"
        >
          pokemon-18
        </li>
        <li
          aria-disabled="false"
          aria-selected="false"
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters MuiAutocomplete-option emotion-14"
          data-option-index="19"
          data-testid="pokemon-pokemon-19"
          id=":r0:-option-19"
          role="option"
          tabindex="-1"
        >
          pokemon-19
        </li>
      </ul>
    </div>
  </div>
</body>
`;
